common/Root.tsx

```diff
@@ -32,33 +31,38 @@ export default class Root extends Component<Props, State> {
   }
 
   public render() {
-    const { store, history } = this.props;
+    const { store } = this.props;
     const { error } = this.state;
 
     if (error) {
       return <ErrorScreen error={error} />;
     }
 
     // key={Math.random()} = hack for HMR from https://github.com/webpack/webpack-dev-server/issues/395
+    const routes = (
+      <div>
+        <Route exact={true} path="/" component={GenerateWallet} />
+        <Route path="/help" component={Help} />
+        <Route path="/swap" component={Swap} />
+        <Route path="/account" component={SendTransaction}>
+          <Route path="send" component={SendTransaction} />
+          <Route path="info" component={SendTransaction} />
+        </Route>
+        <Route path="/send-transaction" component={SendTransaction} />
+        <Route path="/contracts" component={Contracts} />
+        <Route path="/ens" component={ENS} />
+        <Route path="/sign-and-verify-message" component={SignAndVerifyMessage} />
+        <Route path="/pushTx" component={BroadcastTx} />
+        <LegacyRoutes />
+      </div>
+    );
     return (
       <Provider store={store} key={Math.random()}>
-        <Router history={history} key={Math.random()}>
-          <div>
-            <Route exact={true} path="/" component={GenerateWallet} />
-            <Route path="/help" component={Help} />
-            <Route path="/swap" component={Swap} />
-            <Route path="/account" component={SendTransaction}>
-              <Route path="send" component={SendTransaction} />
-              <Route path="info" component={SendTransaction} />
-            </Route>
-            <Route path="/send-transaction" component={SendTransaction} />
-            <Route path="/contracts" component={Contracts} />
-            <Route path="/ens" component={ENS} />
-            <Route path="/sign-and-verify-message" component={SignAndVerifyMessage} />
-            <Route path="/pushTx" component={BroadcastTx} />
-            <LegacyRoutes />
-          </div>
-        </Router>
+        {process.env.HTTPS || process.env.NODE_ENV === 'development' ? (
```

Comment:
I'd like to key in on something more declarative than checking the HTTPS environment variable. Maybe something like `process.env.DOWNLOADABLE` for HashRouter. I could easily see us changing some config related to HTTPS and breaking this part of the build.